lib/aws-elasticache-serverless-stack.ts

```diff
@@ -46,14 +60,15 @@ export class AwsElasticacheServerlessStack extends cdk.Stack {
         subnetIds: elastiCacheSubnetIds,
         kmsKeyId: kmsKey.keyId,
         description: `${props.resourcePrefix}-ElastiCache-Serverless`,
-        majorEngineVersion: "8",
+        majorEngineVersion: props.valkeyEngineVersion,
```

Comment:
**Suggestion:** Add version validation to ensure the provided Valkey engine version is supported by AWS ElastiCache Serverless before deployment.
> 🔥  **High Priority** 
 >[Category: **`Possible Bug`**, Importance: **`8`**]
> **Reason:** Validating the Valkey engine version ensures compatibility with AWS ElastiCache Serverless, preventing deployment failures. This is a valuable addition to improve robustness and reliability, though its impact is slightly less critical than security-related validations.

```suggestion
        if (!['7', '8'].includes(props.valkeyEngineVersion)) {
          throw new Error('Unsupported Valkey engine version. Supported versions are 7 and 8');
        }
        majorEngineVersion: props.valkeyEngineVersion,
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**